src/main/java/com/atlassian/db/replica/spi/ReplicaFailureStrategy.java

```diff
@@ -0,0 +1,14 @@
+package com.atlassian.db.replica.spi;
+
+
+import com.atlassian.db.replica.api.SqlCall;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+public interface ReplicaFailureStrategy {
+    /**
+     * Controls replica failures.
+     */
+    Connection onFailure(SQLException exception, SqlCall<Connection> mainConnection) throws SQLException;
```

Comment:
looking only from the perspective of  `ReplicaFailureStrategy` API it looks that `replicaConnection` is skipped.

Reply:
> looking only from the perspective of ReplicaFailureStrategy API it looks that replicaConnection is skipped.

Replica connection supplier failed and thrown an exception. We have an exception and a supplier for the main connection. What scenario would require a replicaConnection supplier?

Reply:
Retrying to use replicaConnection,